libsolidity/analysis/TypeChecker.cpp

```diff
@@ -323,11 +323,13 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 {
 	if (_function.markedVirtual())
 	{
-		if (_function.annotation().contract->isInterface())
+		if (_function.isConstructor())
+			m_errorReporter.typeError(7001_error, _function.location(), "Constructors cannot be virtual.");
```

Comment:
Was it possible to mark it virtual before? Isn't that a bug in 0.6?

Reply:
It was in `TypeChecker::typeCheckConstructor` before.